pkg/serviceimport/map.go

```diff
@@ -81,19 +105,23 @@ func (m *Map) Put(serviceImport *lighthousev2a1.ServiceImport) {
 		defer m.Unlock()
 
 		remoteService, ok := m.svcMap[key]
+
 		if !ok {
 			remoteService = &serviceInfo{
 				key:        key,
-				clusterIPs: make(map[string]string),
+				clusterIPs: make(map[string][]string),
 				rrCount:    0,
+				isHeadless: serviceImport.Spec.Type == lighthousev2a1.Headless,
 			}
 		}
 
 		for _, info := range serviceImport.Status.Clusters {
-			remoteService.clusterIPs[info.Cluster] = info.IPs[0]
+			remoteService.clusterIPs[info.Cluster] = info.IPs
```

Comment:
I've tested the use case, it works fine if `IPs` is empty list. I had to change some code to make list empty instead of inserting `""` for this reason.